qiskit_experiments/base_experiment.py

```diff
@@ -99,17 +112,22 @@ def run(
         Returns:
             The experiment data object.
         """
-        # Create new experiment data
         if experiment_data is None:
-            experiment_data = self.__experiment_data__(self, backend=backend)
-
-        # Generate and transpile circuits
-        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
+            # Create new experiment data
+            experiment_data = self.__experiment_data__(experiment=self, backend=backend)
+        else:
+            # Override experiment metadata
+            for key, val in self.metadata().items():
+                experiment_data._metadata[key] = val
 
-        # Run circuits on backend
+        # Run options
         run_opts = copy.copy(self.run_options)
         run_opts.update_options(**run_options)
         run_opts = run_opts.__dict__
+        experiment_data._metadata["run_options"] = run_opts
+
+        # Generate and transpile circuits
+        circuits = transpile(self.circuits(backend), backend, **self.transpile_options.__dict__)
```

Comment:
An update of `transpile_options` in the experiment data metadata seems to be missing